parser_library/src/processing/preprocessor.h

```diff
@@ -97,8 +94,12 @@ class preprocessor
 
     static bool is_continued(std::string_view s);
 
-    static void do_highlighting(
-        const semantics::preprocessor_statement_si& stmt, semantics::source_info_processor& src_proc);
+    void do_highlighting(
+        const semantics::preprocessor_statement_si& stmt, semantics::source_info_processor& src_proc) const;
+
+    template<typename PREPROC_STATEMENT, typename ITERATOR>
+    std::shared_ptr<PREPROC_STATEMENT> get_preproc_statement(
+        const std::match_results<ITERATOR>& matches, stmt_part_ids ids, size_t lineno) const;
```

Comment:
does it have to be a member? If not, I suggest moving it into a separate header file + cleaning up the include list.